pg_shard.c

```diff
@@ -329,6 +373,73 @@ PgShardPlanner(Query *query, int cursorOptions, ParamListInfo boundParams)
 
 
 /*
+ * SafeToPushDownAggregate returns true if GROUP BY clause includes the partition column.
+ * This function returns false if the total length of target list is greater than the
+ * total number of columns on the distributed table whose Oid is the second parameter
+ * of this function.
+ */
+static bool
+SafeToPushDownAggregate(Query *query, Oid distributedTableId)
+{
+	List *groupClauseList = query->groupClause;
```

Comment:
Many of these variables can have their scopes reduced (put them in the `else` block).